### ISSUE TITLE
Custom loss function

### ISSUE THREAD

**Issue by ntelo007** [qubvel/segmentation_models#321]
Hi there! 

Thank you for this wonderful repository! My question is:

Can you help me implement a customized loss function and use it with one of your examples?

The loss function that I want to implement completely in Keras (exactly like what you do in your suite) is retrieved from the paper https://arxiv.org/pdf/2003.07311.pdf and can be seen in the following figure:

![image](https://user-images.githubusercontent.com/31326655/79332388-0a313e80-7f1d-11ea-8f52-7509d22e3005.png)

It is written in PyTorch and I don't have the experience to match every PyTorch function with the corresponding Keras version. So far I have written the following:

```
from keras import backend as K

def soft_clDice_loss(gt, pr, iter=50,smooth=1):

    def soft_erode(img):
        p1 = −K.pool2d(−img, (3, 1), (1, 1), (1, 0))
        p2 = −K.pool2d(−img, (1, 3), (1, 1), (0, 1))
        return K.min(p1, p2)

    def soft_dilate(img):
        return K.pool2d(img, (3, 3), (1 ,1), (1, 1))

    def soft_open(img):
        img = soft_erode(img)
        img = soft_dilate(img)
        return img
        
    def soft_skel(img, iters):
        img1 = soft_open(img)
        skel = K.relu(img−img1)

        for j in range(iter):
            img = soft_erode(img)
            img1 = soft_open(img)
            delta = K.relu(img-img1)
            intersect = K.dot(skel, delta)
            skel += K.relu(delta-intersect)
        return skel

    def soft_clDice_loss(iters = 50):
        def loss(y_true, y_pred):
            smooth = 1.
            skel_pred = soft_skel(y_pred, iters)
            skel_true = soft_skel(y_true, iters)
            pres = (K.sum(K.dot(skel_pred, y_true)[:,1:,:,:,:])+smooth)/(K.sum(skel_pred[:,1:,:,:,:])+smooth)    
            rec = (K.sum(K.dot(skel_true, y_pred)[:,1:,:,:,:])+smooth)/(K.sum(skel_true[:,1:,:,:,:])+smooth)    
            cl_dice = 1.- 2.0*(pres*rec)/(pres+rec)
            return cl_dice
        return loss
        
	
##### combine it with regular dice loss
def soft_dice(y_true, y_pred):
    smooth = 1
    intersection = K.sum((y_true * y_pred)[:,1:,:,:,:])
    coeff = (2. *  intersection + smooth) / (K.sum(y_true[:,1:,:,:,:]) + K.sum(y_pred[:,1:,:,:,:]) + smooth)

    return (1. - coeff)

def soft_combined_dice_loss(iters = 15):
    def loss(y_true, y_pred):
        smooth = 1.
        skel_pred = soft_skel(y_pred, iters)
        skel_true = soft_skel(y_true, iters)
        pres = (K.sum(tf.math.multiply(skel_pred, y_true)[:,1:,:,:,:])+smooth)/(K.sum(skel_pred[:,1:,:,:,:])+smooth)    
        rec = (K.sum(tf.math.multiply(skel_true, y_pred)[:,1:,:,:,:])+smooth)/(K.sum(skel_true[:,1:,:,:,:])+smooth)    
        cl_dice = 1.- 2.0*(pres*rec)/(pres+rec)
        dice = soft_dice(y_true, y_pred)
        return 0.5*dice+0.5*cl_dice
    return loss
```

As you can see I am also interested to combine it with a soft Dice loss. Could this loss function be combined with your implementation of the Dice loss?
